flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -402,25 +401,4 @@ public void run() {
 			}
 		}
 	}
-
-	private static final class MetricRegistryThreadFactory implements ThreadFactory {
-		private final ThreadGroup group;
-		private final AtomicInteger threadNumber = new AtomicInteger(1);
-
-		MetricRegistryThreadFactory() {
-			SecurityManager s = System.getSecurityManager();
-			group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
-		}
-
-		public Thread newThread(Runnable r) {
-			Thread t = new Thread(group, r, "Flink-MetricRegistry-" + threadNumber.getAndIncrement(), 0);
```

Comment:
The thread name will be different after this change. New thread naming scheme is `<poolName>-thread-<threadNumber>`.

Reply:
That's true. It's not important imo; in fact I would argue that it's actually an argument for merging it to make the names more consistent.